### ISSUE TITLE
Add NVIDIA GPU local and remote attestation

### ISSUE THREAD

**Issue by zvonkok** [confidential-containers/guest-components#550]
Using this issue to track the attestation changes for supporting NVIDIA GPUs with our attestation_sdk: https://github.com/NVIDIA/nvtrust, lets start with the obvious: 

- [ ] Attester
- [ ] Verifer

There are more things to come, maybe a KMS backend as well.

**Comment by imlk0** [comment]
Hi @zvonkok, it's a nice work. I'm curious if this will mean implementing a new Attester in AA crate and a new Verifier in AS crate? Also, I'm wondering if it will use something like [nvml-wrapper](https://crates.io/crates/nvml-wrapper) under the hood or just use ffi to call the attestation_sdk python library?

**Comment by zvonkok** [comment]
@imlk0 Yes new Attester in AA create and a new Verifier in AS crate. No not using nvml-wrapper this is just a small part of the attestation_sdk. I would need to implement a ton of things to make this work just by using nvml-wrapper. 
As a start to see something working maybe just calling an binary (PyInstaller). 
We can see from there if we need attestation_sdk updates or FFI lets see but I hope to get enough reviews to move this in the right direction.

**Comment by imlk0** [comment]
> @imlk0 Yes new Attester in AA create and a new Verifier in AS crate. No not using nvml-wrapper this is just a small part of the attestation_sdk. I would need to implement a ton of things to make this work just by using nvml-wrapper. As a start to see something working maybe just calling an binary (PyInstaller). We can see from there if we need attestation_sdk updates or FFI lets see but I hope to get enough reviews to move this in the right direction.

Thanks for your reply.

**Comment by 1570005763** [comment]
Great proposal! I'm curious about the part regarding KMS.
Could you perhaps share the needs for KMS in NVIDIA GPU remote attestation? Currently, CDH has a dedicated KMS plugin interface, which can interface with various KMS. Do you think this feature might be applicable or beneficial in your work?
